src/wazuh_modules/vulnerability_scanner/tests/unit/scanOrchestrator_test.cpp

```diff
@@ -677,5 +782,5 @@ TEST_F(ScanOrchestratorTest, TestRunOrchestrationThatGeneratesException)
             EXPECT_STREQ(va_arg(args, const char*), "Test exception");
         });
 
-    EXPECT_NO_THROW(scanOrchestrator.run(syscollectorDelta));
+    EXPECT_THROW(scanOrchestrator.run(syscollectorDelta), std::runtime_error);
```

Comment:
Because generates an exception

```test
[ RUN      ] ScanOrchestratorTest.TestRunOrchestrationThatGeneratesException
src/wazuh_modules/vulnerability_scanner/tests/unit/scanOrchestrator_test.cpp:785: Failure
Expected: scanOrchestrator.run(syscollectorDelta) doesn't throw an exception.
  Actual: it throws std::runtime_error with description "Test exception".
[  FAILED  ] ScanOrchestratorTest.TestRunOrchestrationThatGeneratesException (94 ms)
```

Possible this is due a validation added in the facade class

Reply:
Doing deeper research, this test is made to throw an exception, and the new change is the correct way.

```cpp
auto mockThrowException = [&](std::shared_ptr<TScanContext<TrampolineOsDataCache>> spScanContext)
        -> std::shared_ptr<TScanContext<TrampolineOsDataCache>>
    {
        throw std::runtime_error {"Test exception"};
    };
```

Reply:
The test was removed at #21456.

The name of the test suggests that an exception will be thrown but it was previously cached  by a block that was removed. That is why you had to change it but now it might not have too much sense.
This can be solved in the rebase I suppose.

Reply:
I agree, the rebase will be deleted, in order to pass CI/CD we kept this change for now.